### DISCUSSION TITLE
Combined Illumina libraries of different length

### DISCUSSION THREAD

**Discussion by Phismil** [ablab/spades#1060] [Q&A]
Dear Spades developers/users
I have inherited a dataset of Illumina PE 150 and PE 250. I am wondering if it is feasible or recommended to run different length libraries together in a single run?
I usually assemble 2 x 150 first and then use the assembled contigs as "trusted contigs" in only 2x250 assembly.
I appreciate your recommendation in advance

**Comment by asl** [comment]
> I usually assemble 2 x 150 first and then use the assembled contigs as "trusted contigs" in only 2x250 assembly.

This is a bit strange "protocol". Where you obtained it? SPAdes happily supports multiple input libraries.

**Reply by Phismil** [reply]
Thank you Anton for your response. I am aware that Spades can handle multiple libraries, but I did not know that It can also support multiple libraries of different length.That makes the assembly far faster. Cheers

**Reply by asl** [reply]
Yes, there are no limits on the lengths. I would suggest to specify the library with smaller insert size first

**Comment by SergeyBaikal** [comment]
I have two libraries: 2x300 Illumina and 2x150 MGI. I assembled them into one assembly by metaSPAdes (cross-assembly). I used the cat command to merge them (R1_1+R1_2 and R2_1+R2_2). There were no errors or warnings in the log file.
